src/helpers/handleCameraFOV.tsx

```diff
@@ -0,0 +1,45 @@
+'use client';
+import { useThree, useFrame } from '@react-three/fiber';
+import { useEffect } from 'react';
+import { PerspectiveCamera } from 'three';
+
+interface AdjustCameraProps {
+  mobileFov: number;
+  desktopFov: number;
+  aspectRatio: number;
+}
+
+export default function AdjustCamera({
+  mobileFov,
+  desktopFov,
+  aspectRatio,
+}: AdjustCameraProps) {
+  const { camera } = useThree();
+
+  useEffect(() => {
+    const handleResize = () => {
+      if (camera instanceof PerspectiveCamera) {
+        const aspect = window.innerWidth / window.innerHeight;
+        camera.aspect = aspect;
+        camera.fov = aspect > aspectRatio ? desktopFov : mobileFov; // Adjust FOV based on aspect ratio
+        camera.updateProjectionMatrix();
+      }
+    };
+
+    handleResize(); // Initial call
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('resize', handleResize);
+    };
+  }, [camera]);
+
+  useFrame(() => {
```

Comment:
may be useFrame is redundant as we are updating matrix in useEffect?

Reply:
Yes, removed the useFrame.